bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolBridgeHandler.java

```diff
@@ -0,0 +1,345 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.resol.handler;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.resol.internal.ResolBindingConstants;
+import org.openhab.binding.resol.internal.ResolBridgeConfiguration;
+import org.openhab.binding.resol.internal.discovery.ResolDeviceDiscoveryService;
+import org.openhab.core.i18n.LocaleProvider;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.resol.vbus.Connection;
+import de.resol.vbus.Connection.ConnectionState;
+import de.resol.vbus.ConnectionAdapter;
+import de.resol.vbus.Packet;
+import de.resol.vbus.Specification;
+import de.resol.vbus.SpecificationFile;
+import de.resol.vbus.SpecificationFile.Language;
+import de.resol.vbus.TcpDataSource;
+import de.resol.vbus.TcpDataSourceProvider;
+
+/**
+ * The {@link ResolBridgeHandler} class handles the connection to the VBUS/LAN adapter.
+ *
+ * @author Raphael Mack - Initial contribution
+ */
+@NonNullByDefault
+public class ResolBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(ResolBridgeHandler.class);
+
+    private Language lang;
+    private Locale locale;
+
+    private String ipAddress = "";
+    private String password = "";
+    private int refreshInterval;
+    private boolean isConnected = false;
+    private String unconnectedReason = "";
+
+    // Background Runnable
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private @Nullable Connection tcpConnection;
+    private Specification spec;
+
+    // Managing Thing Discovery Service
+    private @Nullable ResolDeviceDiscoveryService discoveryService = null;
+
+    private boolean scanning;
+
+    public ResolBridgeHandler(Bridge bridge, @Nullable LocaleProvider localeProvider) {
+        super(bridge);
+        spec = Specification.getDefaultSpecification();
+
+        if (localeProvider != null) {
+            locale = localeProvider.getLocale();
+            lang = SpecificationFile.getLanguageForLocale(getLocale());
+
+        } else {
+            locale = Locale.getDefault();
+            lang = Language.En;
+        }
```

Comment:
Is it worth to increase the depth of the call tree for each received packet for checking the locale only for this rather seldom use case? As a user, I personally would favor runtime performance over this. Or is there at least a callback that I can register to get notified on the locale update?

Reply:
The performance impact would be negligible and would allow the user to change the locale in openhab without having to restart your binding.
That said I don't know what the performance impact of calling `SpecificationFile.getLanguageForLocale` would be. So it might be worth caching the return value from that method and only changing it when the returned locale from the provider changes.

Reply:
`SpecificationFile.getLanguageForLocale` looks also quite fast, so I changed to read the locale dynamically.